packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -31,39 +39,76 @@ export abstract class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
       : '';
     const execFilePathOrPaths = response.data;
     const execution = new CliCommandExecutor(this.build(response.data), {
-      cwd: workspacePath
+      cwd: workspacePath,
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
     }).execute(cancellationToken);
 
-    let stdErr = '';
-    execution.stderrSubject.subscribe(realData => {
-      stdErr += realData.toString();
+    channelService.streamCommandStartStop(execution);
+    channelService.showChannelOutput();
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(realData => {
+      stdOut += realData.toString();
     });
 
     execution.processExitSubject.subscribe(async exitCode => {
       this.logMetric(execution.command.logName, startTime);
-      if (exitCode !== 0) {
-        try {
-          const deployErrorParser = new ForceDeployErrorParser();
-          const fileErrors = deployErrorParser.parse(stdErr);
+      try {
+        const deployParser = new ForceDeployResultParser(stdOut);
+        const errors = deployParser.getErrors();
+        if (errors) {
           handleDiagnosticErrors(
-            fileErrors,
+            errors,
             workspacePath,
             execFilePathOrPaths,
             ForceSourceDeployExecutor.errorCollection
           );
-        } catch (e) {
-          telemetryService.sendError(
-            'Error while creating diagnostics for vscode problem view.'
-          );
-          console.error(
-            'Error while creating diagnostics for vscode problem view.'
-          );
+        } else {
+          ForceSourceDeployExecutor.errorCollection.clear();
         }
-      } else {
-        ForceSourceDeployExecutor.errorCollection.clear();
+        this.outputResult(deployParser);
+      } catch (e) {
+        telemetryService.sendError(
+          'Error while creating diagnostics for vscode problem view.'
+        );
+        console.error(
+          'Error while creating diagnostics for vscode problem view.'
+        );
       }
     });
 
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private outputResult(parser: ForceDeployResultParser) {
+    const table = new Table();
+    const errors = parser.getErrors();
+    const successes = parser.getSuccesses();
+    const deployedSource = successes ? successes.result.deployedSource : undefined;
+
+    if (deployedSource) {
+      const outputTable = table.createTable(deployedSource as unknown as Row[], [
```

Comment:
`deployedSource as unknown as Row[]` is strange looking, but it's how to force cast. Because the Row type is dynamic with no specific keys, this will work without having to validate the keys of deployedSource.

Reply:
Awesome, it's really cool that this can be extended and reused for source:push results too.